service/wechat/wechat.go

```diff
@@ -0,0 +1,146 @@
+package wechat
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"github.com/pkg/errors"
+	"github.com/silenceper/wechat/v2"
+	"github.com/silenceper/wechat/v2/cache"
+	"github.com/silenceper/wechat/v2/officialaccount/config"
+	"github.com/silenceper/wechat/v2/officialaccount/message"
+	"github.com/silenceper/wechat/v2/util"
+)
+
+// Config is the Service configuration.
+type Config struct {
+	AppID          string
+	AppSecret      string
+	Token          string
+	EncodingAESKey string
+	Cache          cache.Cache
+}
+
+// wechatMessageManager abstracts go-wechat's message.Manager for writing unit tests
+type wechatMessageManager interface {
+	Send(msg *message.CustomerMessage) error
+}
+
+// Service encapsulates the WeChat client along with internal state for storing users.
+type Service struct {
+	config         *Config
+	messageManager wechatMessageManager
+	userIDs        []string
+}
+
+// New returns a new instance of a WeChat notification service.
+func New(cfg *Config) *Service {
+	wc := wechat.NewWechat()
+	wcCfg := &config.Config{
+		AppID:          cfg.AppID,
+		AppSecret:      cfg.AppSecret,
+		Token:          cfg.Token,
+		EncodingAESKey: cfg.EncodingAESKey,
+		Cache:          cfg.Cache,
+	}
+
+	oa := wc.GetOfficialAccount(wcCfg)
+
+	return &Service{
+		config:         cfg,
+		messageManager: oa.GetCustomerMessageManager(),
+	}
+}
+
+// WaitForOneOffVerification waits for the verification call from the WeChat backend.
+//
+// Should be running when (re-)applying settings in wechat configuration.
+//
+// Set devMode to true when using the sandbox.
+//
+// See https://developers.weixin.qq.com/doc/offiaccount/en/Basic_Information/Access_Overview.html
+func (s *Service) WaitForOneOffVerification(
+	addr string,
+	devMode bool,
+	callback func(r *http.Request, verified bool)) error {
+	srv := &http.Server{Addr: addr}
```

Comment:
```suggestion
func (s *Service) WaitForOneOffVerification(serverURL string, devMode bool, callback verificationCallbackFunc) error {
	srv := &http.Server{Addr: serverURL}
```

I know. Introducing the *verificationCallbackFunc* type is not necessary but I just prefer the clean look of the function signature and I think reading it that way makes it easier to understand what the callback actually is for. 

Additionally *serverURL* is clearer in this context than just *addr*. If this would be a function like *NewServer* or something the like, then *addr* would be expressive enough but in this context its not.